data_structures/arrays/sparse_table.py

```diff
@@ -0,0 +1,94 @@
+"""
+    Sparse table is a data structure that allows answering range queries on
+    a static array, i.e. the elements do not change throughout all the queries.
+
+    The implementation below will solve the problem of Range Minimum Query:
+    Finding the minimum value of a subset [L..R] of a static array.
+
+    Overall time complexity: O(nlogn)
+    Overall space complexity: O(nlogn)
+
+    Wikipedia link: https://en.wikipedia.org/wiki/Range_minimum_query
+"""
+
+
+def build_sparse_table(arr: list[int], arr_length: int) -> list[list[int]]:
```

Comment:
Let's lose `arr_length` because it can be computed via `len(arr)`.
Please try to come up with a better name than `arr`.  Are these options, items, choices, candidates?